src/main/java/lotto/domain/Rank.java

```diff
@@ -1,63 +1,103 @@
 package lotto.domain;
 
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
+import java.util.function.Function;
+
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toMap;
+import static lotto.domain.MatchBonus.*;
 
 public enum Rank {
-    FIRST(6, 2_000_000_000),
-    SECOND(5, 30_000_000),
-    THIRD(5, 1_500_000),
-    FOURTH(4, 50_000),
-    FIFTH(3, 5_000),
-    MISS(0,0);
-
-    private final int countOfMatch;
-    private final int winningMoney;
-    private final static Map<MatchBonus, Rank> RANK_BY_MATCH_COUNT = new HashMap<MatchBonus, Rank>() {
-        {
-            put(MatchBonus.of(0, true), MISS);
-            put(MatchBonus.of(0, false), MISS);
-            put(MatchBonus.of(1, true), MISS);
-            put(MatchBonus.of(1, false), MISS);
-            put(MatchBonus.of(2, true), MISS);
-            put(MatchBonus.of(2, false), MISS);
-            put(MatchBonus.of(3, true), FIFTH);
-            put(MatchBonus.of(3, false), FIFTH);
-            put(MatchBonus.of(4, true), FOURTH);
-            put(MatchBonus.of(4, false), FOURTH);
-            put(MatchBonus.of(5, true), SECOND);
-            put(MatchBonus.of(5, false), THIRD);
-            put(MatchBonus.of(6, true), FIRST);
-            put(MatchBonus.of(6, false), FIRST);
+    FIRST(MATCH_COUNT_BY_MATCH_BONUS_FALSE.get(6), 2_000_000_000),
+    SECOND(MATCH_COUNT_BY_MATCH_BONUS_TRUE.get(5), 30_000_000),
+    THIRD(MATCH_COUNT_BY_MATCH_BONUS_FALSE.get(5), 1_500_000),
+    FOURTH(MATCH_COUNT_BY_MATCH_BONUS_FALSE.get(4), 50_000),
+    FIFTH(MATCH_COUNT_BY_MATCH_BONUS_FALSE.get(3), 5_000),
+    MISS(MATCH_COUNT_MISS, 0);
 
-        }
-    };
+    private MatchBonus matchBonus;
+    private final int winningMoney;
+    private final static Map<MatchBonus, Rank> RANK_BY_MATCH_COUNT =
+            Arrays.stream(values())
+                    .filter(rank -> rank != Rank.MISS)
+                    .collect(collectingAndThen(
+                            toMap(Rank::getMatchBonus, Function.identity()),
+                            Collections::unmodifiableMap));
 
-    Rank(int countOfMatch, int winningMoney) {
-        this.countOfMatch = countOfMatch;
+    Rank(MatchBonus matchBonus, int winningMoney) {
+        this.matchBonus = matchBonus;
         this.winningMoney = winningMoney;
     }
 
-    public int getCountOfMatch() {
-        return countOfMatch;
+    public MatchBonus getMatchBonus() {
+        return matchBonus;
     }
 
     public int getWinningMoney() {
         return winningMoney;
     }
 
     public static Rank valueOf(int countOfMatch, boolean matchBonus) {
-        return RANK_BY_MATCH_COUNT.get(MatchBonus.of(countOfMatch,matchBonus));
+        Rank rank = RANK_BY_MATCH_COUNT.get(MatchBonus.of(countOfMatch, matchBonus));
+        return rank == null ? MISS : rank;
```

Comment:
```suggestion
        return RANK_BY_MATCH_COUNT.getOrDefault(MatchBonus.of(countOfMatch, matchBonus), MISS);
```

getOrDefault를 활용해보세요.

Reply:
Map쓸 때 getOrDefault 항상 염두하는 습관 들이겠습니다!